tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-type-narrowing-negative.bal

```diff
@@ -0,0 +1,505 @@
+        select vwFn(item);
+
+    _ = from var item in data
+        where item !is V|W
+        select yzFn(item); // error incompatible types: expected '(Y|Z)', found '(W|Y|Z)'
+
+    check from var item in data
+        where item !is V|W
+        do {
+            yzFn(item); // error incompatible types: expected '(Y|Z)', found '(W|Y|Z)'
+        };
+
+    _ = from var item in data
+        where item is V|W|Y
+        select vwyFn(item);
+
+    _ = from var item in data
+        where item !is V|W|Y
+        select zFn(item);
+}
+
+type Q record {|
+    string a;
+|};
+
+type R record {
+    int b;
+};
+
+type T record {
+    boolean a;
+};
+
+function qFn(Q x) {
+}
+
+function rFn(R x) {
+}
+
+function rtFn(R|T x) {
+}
+
+function test10((Q|R)[] data1, (Q|R|T)[] data2) returns error? {
+    _ = from var item in data1
+        where item is Q
+        select qFn(item);
+
+    _ = from var item in data1
+        where item !is Q
+        select rFn(item); // error incompatible types: expected 'R', found '(Q|R)'
+
+    check from var item in data1
+        where item !is Q
+        do {
+            rFn(item); // error incompatible types: expected 'R', found '(Q|R)'
+        };
+
+    _ = from var item in data2
+        where item is Q
+        select qFn(item);
+
+    _ = from var item in data2
+        where item !is Q
+        select rtFn(item); // error incompatible types: expected '(R|T)', found '(Q|R|T)'
+}
+
+type Integers 1|2|3;
+
+type Chars "A"|"B"|"C";
+
+function test11(Integers[] numbers, Chars[] chars) returns error? {
+    1[] _ = from int item in numbers
+        where item == 1
+        select item;
+
+    (1|2)[] _ = from int item in numbers
+        where item == 1 || item == 2
+        select item;
+
+    check from Integers item in numbers
+        where item == 1 || item == 2
+        do {
+            1 _ = item; // error incompatible types: expected '1', found '(1|2)'
+        };
+
+    int[] _ = from int item in numbers
+        where item is 1
+        select item * 2;
+
+    "C"[] _ = from Chars item in chars
+        where item == "C"
+        select item;
+
+    check from Chars item in chars
+        where item == "C"
+        do {
+            "C" _ = item;
+        };
+
+    int?[] a = [1, 2, 3];
+    var _ = from var item in a
+        where item == 1
+        let () varName = item // error incompatible types: expected '()', found '1'
+        select item;
+
+    1[] _ = from var item in 1 ... 3
+        let int? one = 1
+        where one == 1
+        select one;
+
+    check from var item in 1 ... 3
+        let int? one = 1
+        where one == 1
+        do {
+            2 two = one; // error incompatible types: expected '2', found '1'
+        };
+
+    (int|string)[] arr = [1, 2, 3];
+    check from int|string item in arr
+        where item is int
+        let int num = item
+        do {
+            int _ = num * 2;
+        };
+
+    //    Should be enabled once issue #32271 is fixed
```

Comment:
Shall we add comments on the issues to enable these?

Reply:
Added. thanks